inc/namespace.php

```diff
@@ -196,3 +198,77 @@ function enqueue_scripts() {
 	// Print queued scripts.
 	print_head_scripts();
 }
+
+/**
+ * Schedule index maintenance daily.
+ */
+if( ! wp_next_scheduled( __NAMESPACE__ . '\\index_maintenance' ) ) {
+	wp_schedule_event( time(), 'daily',  __NAMESPACE__ . '\\index_maintenance' );
+}
+
+
+/**
+ * Delete old Analytics indexes in Elasticsearch.
+ */
+function delete_old_indexes() : ?array {
+	// Index age in days.
+	$duration = apply_filters( 'altis.analytics.index_age', 7 );
+
+	// If duration has been set out of the 60-day limit, default to 7 days and warn user.
+	if ( $duration < 1 || $duration > 60 ) {
+		$duration = 7;
+		trigger_error( 
+			'Analytics data retention period must be between 1 and 60 days, defaulting to 7.', 
+			E_USER_WARNING 
+		);
+	}
+
+	// Get index name by date. 
+	$date = new \DateTime();
+	$date->sub( new \DateInterval( 'P'. $duration . 'D' ) );
+	$index_date = ( $date->format('U') );
+
+	// Get URL.
+	$index_list = wp_remote_request( get_elasticsearch_url() . '/analytics-*/');
+
+	$indices = json_decode( wp_remote_retrieve_body( $index_list ), true);
```

Comment:
Need to check whether the response was successful here first.
```suggestion
	$indices = json_decode( wp_remote_retrieve_body( $index_list ), true );
```